### PR TITLE
Remove support for legacy dav endpoint

### PR DESCRIPTION
**Pull request by TheOneRing**

### CODE REVIEW

**Review by michaelstingl — APPROVED** [review]
Deprecation needs to be announced in 2.10.1 release communication.

File: changelog/unreleased/9538 [hunk 1]
```diff
@@ -0,0 +1,5 @@
+Enhancement: Remove use of legacy dav endpoint
+
+We no longer guess the dav endpoint depending on the chnunking ng feature.
```
Comment (fmoc) [inline changelog/unreleased/9538:3]:
```suggestion
Enhancement: Remove use of legacy DAV endpoint

We no longer guess the DAV endpoint depending on the chunking-ng feature.
```

**Review by fmoc — APPROVED** [review]

**Comment by ownclouders** [comment]
Results for <strong>GUI-tests</strong> https://drone.owncloud.com/owncloud/client/11152/5/1
GUI Logs: (https://cache.owncloud.com/public/owncloud/client/11152/guiReportUpload/index.html)
Server Logs: (https://cache.owncloud.com/public/owncloud/client/11152/guiReportUpload/serverlog.log)

**Comment by ownclouders** [comment]
Results for <strong>GUI-tests</strong> https://drone.owncloud.com/owncloud/client/11154/5/1
GUI Logs: (https://cache.owncloud.com/public/owncloud/client/11154/guiReportUpload/index.html)
Server Logs: (https://cache.owncloud.com/public/owncloud/client/11154/guiReportUpload/serverlog.log)

**Pull request merged**